### PR TITLE
Implement user confirmation in clear command

### LINKED ISSUE #11: Delete/Clear all students
As a TA, I can delete all students from TAPA, so that I can remove students that were taught in previous semesters and are no longer relevant.

### PR DESCRIPTION
**Pull request by brelkh**
This PR extends the functionality of the clear command. Before clearing all the students in TAPA, TAPA will request for confirmation from the user first.

After several failed attempts at implementing this, I found that the most effective and seamless way was to utilise a new `ConfirmClearCommand` class. 

**Rough outline of sequence:**
1. User inputs "`clear`"
2. `ClearCommand` object is created which returns a `CommandResult` with field `isClearRequest` set to `true`.
3. `MainWindow` observes `isClearRequest` is set to true and changes the `CommandBox` from one that parses all commands to one that only recognises `confirm` input (all other inputs will return an "Unknown command" message).
4. If a user enters `confirm`, all students are cleared from TAPA. Else, the `CommandBox` is reverted back to the one that can parse all commands.

image
image
image

**Bugs/Potential improvements:**
1. A minor bug is that when the user inputs `clear`, they have to click the command box to continue typing commands.
2. When implementing a `CommandBox` that only recognises `confirm`, some of the code is duplicated since it is quite similar to the `CommandBox` that recognises all commands. While these chunks of code could be abstracted into a separate class, I think that would be quite difficult and could be done later on if we need to touch up on code quality.
3. Perhaps this confirmation feature can be upgraded to a button in the future (though that will be more difficult to test).

**Notes:**
Closes #11.
Testcases have yet to be implemented.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ConfirmClearCommand.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import seedu.address.model.AddressBook;
+import seedu.address.model.Model;
+
+/**
+ * Clears the address book.
+ */
+public class ConfirmClearCommand extends Command {
+
+    public static final String COMMAND_WORD = "confirm";
+    public static final String MESSAGE_CLEAR_ACKNOWLEDGEMENT = "Address book completely cleared, as requested!";
```
Comment (Korochii) [inline src/main/java/seedu/address/logic/commands/ConfirmClearCommand.java:14]:
```suggestion
    public static final String MESSAGE_CLEAR_ACKNOWLEDGEMENT = "TAPA has been completely cleared, as requested!";
```

File: src/main/java/seedu/address/logic/commands/ConfirmClearCommand.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.model.AddressBook;
+import seedu.address.model.Model;
+
+/**
+ * Clears the address book.
```
Comment (Korochii) [inline src/main/java/seedu/address/logic/commands/ConfirmClearCommand.java:9]:
```suggestion
 * Clears TAPA.
```

File: src/main/java/seedu/address/logic/commands/ClearCommand.java [hunk 3]
```diff
@@ -2,22 +2,19 @@
 
 import static java.util.Objects.requireNonNull;
 
-import seedu.address.model.AddressBook;
 import seedu.address.model.Model;
 
 /**
- * Clears the address book.
+ * Requests for confirmation from the user to clear the address book.
```
Comment (Korochii) [inline src/main/java/seedu/address/logic/commands/ClearCommand.java:8]:
```suggestion
 * Requests for confirmation from the user to clear TAPA.
```

**Review by Korochii — APPROVED** [review]
Overall LGTM!
This is indeed an interesting and working approach in extending the existing `clear` command.
Good job! 😃 

(Abit of a nitpicking, but I feel that it might be better if we replace "address book" with "TAPA" in the comments...)

**Comment by Korochii** [comment]
> **Bugs/Potential improvements:**
> 
>     1. A minor bug is that when the user inputs `clear`, they have to click the command box to continue typing commands.
> 
>     2. When implementing a `CommandBox` that only recognises `confirm`, some of the code is duplicated since it is quite similar to the `CommandBox` that recognises all commands. While these chunks of code could be abstracted into a separate class, I think that would be quite difficult and could be done later on if we need to touch up on code quality.
> 
>     3. Perhaps this confirmation feature can be upgraded to a button in the future (though that will be more difficult to test).

Perhaps we can create an issue for this and work on it if we have time?

**Comment by brelkh** [comment]
> Perhaps we can create an issue for this and work on it if we have time?

@Korochii Alright, that sounds good to me 👍 
As for the renaming of address book to TAPA, thanks and I've committed your suggestions!

**Pull request merged**